components/Authentication/signup.jsx

```diff
@@ -115,7 +118,151 @@ const Signup = ({ navigation }) => {
+                                {reps: 12, weight: 100},
+                              ]
+                            }
+                          ]
+            }
+
+            const legsWorkout = {
+              name: "Legs",
+              muscleGroups: "Quads Glutes",
+              lastPerformed: currentTime,
+              exercises: [
+                            {
+                              exerciseID: "31ROy02NIqplIBvXoaeB", 
+                              sets: [
+                                {time: 10, weight: 135},
+                                {time: 10, weight: 135},
+                                {time: 10, weight: 135},
+                              ]},
+                            {
+                              exerciseID: "2TvJvGO8CuxXzxk1D2Si",
+                              sets: [
+                                {reps: 10, weight: 100},
+                                {reps: 10, weight: 100},
+                                {reps: 10, weight: 100},
+                              ]
+                            },
+                            {
+                              exerciseID: "XzkLscitllVWr1sRrMAk",
+                              sets: [
+                                {reps: 10, weight: 100},
+                                {reps: 10, weight: 100},
+                                {reps: 10, weight: 100},
+                              ]
+                            }
+                          ]
+            }
+
+            //set initial user 
+            const userRef = db.collection('users').doc(user.user.uid);
+            
+            const userDataRef = userRef.set(userData);
+            const userWorkout1 = userRef.collection('workouts').doc();
+            const userWorkout2 = userRef.collection('workouts').doc();
+            const userWorkout3 = userRef.collection('workouts').doc();
+
+            Promise.all(
+              userDataRef,
+              userWorkout1.set(pushWorkout), 
+              userWorkout2.set(pullWorkout),
+              userWorkout3.set(legsWorkout)
+            ).then(() => {
+              console.log("workouts and user set successfully.");
+            }).catch((error) => {
+              console.log(error);
+            });
+
+            const cycleData = {
+              name: 'Push, Pull, Legs',
+              workoutIDs: [userWorkout1.id, userWorkout2.id, userWorkout3.id],
+            }
+
+            const cyclesDataRef = userRef.collection('cycles').doc();
+            
+            cyclesDataRef.set(cycleData)
+              .then(() => {
+                console.log("cycles set successfully.");
+              }).catch((error) => {
+                console.log(error);
+              });
+
+            userRef.update({selectedCycleId: cyclesDataRef.id})
+              .then(() => {
+                console.log("selected cycle updated.")
+              }).catch((error) => {
+                console.log(error);
+              });
```

Comment:
Could we do these in parallel as well?

Reply:
In parallel with each other or in parallel with all of the other ones?

Reply:
At least with each other, optimally it'd be with the other stuff but I'm not sure if we can do those together

Reply:
Right I think I have that working in my updated code, I don't think it's possible to get them to all run in parallel tho